lumicks/pylake/kymotracker/kymoline.py

```diff
@@ -416,3 +416,49 @@ def fit_binding_times(self, n_components):
         return _kinetic_mle_optimize(
             n_components, dwelltimes_sec, min_observation_time, max_observation_time
         )
+
+    def _histogram_binding_events(self, kind, bins=10):
+        """Make histogram of bound events.
+
+        Parameters
+        ----------
+        kind : {'binding', 'all'}
+            Type of events to count. 'binding' counts only the first position of each track whereas
+            'all' counts positions of all time points in each track.
+        bins : int or sequence of scalars
+            Definition of the histogram bins; passed to `np.histogram()`. If `int`, defines the total
+            number of bins. If sequence of scalars, defines the edges of the bins.
+        """
+        if kind == "all":
+            slc = slice(None)
+        elif kind == "binding":
+            slc = slice(0, 1)
+        else:
+            raise ValueError(f"`kind` argument '{kind}' must be 'all' or 'binding'.")
+        binds = np.hstack([track.position[slc] for track in self])
+
+        counts, edges = np.histogram(binds, bins=bins)
+        width = np.diff(edges[:2])[0]
+        centers = edges[:-1] + width / 2
+        return counts, centers, width
+
+    def plot_binding_histogram(self, kind, bins=10, **kwargs):
+        """Make histogram of bound events.
+
+        Parameters
+        ----------
+        kind : {'binding', 'all'}
+            Type of events to count. 'binding' counts only the first position of each track whereas
+            'all' counts positions of all time points in each track.
+        bins : int or sequence of scalars
+            Definition of the histogram bins; passed to `np.histogram()`. If `int`, defines the total
+            number of bins. If sequence of scalars, defines the edges of the bins.
+        **kwargs
+            Keyword arguments forwarded to `plt.bar()`.
+        """
+        import matplotlib.pyplot as plt
```

Comment:
I'd probably move imports to the top, as explained [here](https://stackoverflow.com/a/9614212/3154670)

Reply:
I'd agree, although generally throughout the library we keep `matplotlib` imports within the functions where they are needed. This is because plotting is only required for a select number of convenience functions and so only imported specifically when needed. This probably doesn't lead to a huge boost in import time for `pylake` in general, but here, for consistency, I used the same pattern we usually do